app/src/main/java/edu/agh/susgame/front/ui/component/game/map/GameGraphComponent.kt

```diff
@@ -0,0 +1,148 @@
+
+@Composable
+internal fun GameGraphComponent(mapState: GameGraph) {
+    val changeNodeInfoVisibility = remember { mutableStateOf(false) }
+    val nodeToShow = remember { mutableIntStateOf(0) }
+    val zoomState = remember {
+        ZoomState(
+            maxZoomIn = 2f,
+            maxZoomOut = 0.5f,
+            totalSize = mapState.mapSize,
+        )
+    }
+
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .fillMaxHeight()
+            .clipToBounds()
+            .pointerInput(Unit) {
+                detectTransformGestures { _, pan, zoom, _ ->
+                    zoomState.scale(zoom)
+                    zoomState.move(pan)
+                }
+            }
+            .background(Color.Green)
+    ) {
+        Box(
+            modifier = Modifier
+                .graphicsLayer(
+                    scaleX = zoomState.scaleValue(),
+                    scaleY = zoomState.scaleValue(),
+                    translationX = zoomState.translationX(),
+                    translationY = zoomState.translationY(),
+                    clip = false
+                )
+        ) {
+            Canvas(modifier = Modifier.fillMaxSize()) {// it will be improved, just testing how drawing works
+                mapState.edges.forEach { (_, edge) ->
+
+                    val startXY = mapState.nodes[edge.firstNodeId]
+                    val endXY = mapState.nodes[edge.secondNodeId]
+                    if (startXY != null && endXY != null) {
+                        drawLine(
+                            color = Color.Black,
+                            start = Offset(
+                                startXY.position.x.dp.toPx(),
+                                startXY.position.y.dp.toPx(),
+                            ),
+                            end = Offset(
+                                endXY.position.x.dp.toPx(),
+                                endXY.position.y.dp.toPx(),
+                            ),
+                            strokeWidth = 3f
+                        )
+                    }
+                }
+            }
+
+            Text(zoomState.scaleValue().toString(), color = Color.Black)
+            mapState.nodes.forEach { (key, node) ->
+                Button(
+                    modifier = Modifier
+                        .offset(
+                            x = (node.position.x - buttonWidth / 2).dp,
+                            y = (node.position.y - buttonHeight / 2).dp,
+                        )
+                        .size(
+                            width = buttonWidth.dp,
+                            height = buttonHeight.dp
+                        ),
+                    onClick = {
+                        changeNodeInfoVisibility.value = true
+                        nodeToShow.intValue = node.id
+                    },
+                ) {
+                    Text(
+                        text = node.name + ":" + key,
```

Comment:
Byłbym zwolennikiem, żeby tutaj też nie dawać tego `key`, czyli `id`, bo to bardziej myli, niż pomaga

Teraz tak mnie naszło, że jeśli te przyciski będą obrazkami, to dodawanie tekstu może nie być takie łatwe. Może będzie się to dało jakoś ograć, żeby wyświetlał się pod przyciskiem. Na razie niech zostanie jak jest